build-emacs-for-macos

```diff
@@ -456,6 +456,9 @@ class Build
       if options[:tree_sitter] && supports_tree_sitter?
         configure_flags << '--with-tree-sitter'
       end
+      if options[:fd_setsize]
+        configure_flags << "CFLAGS=-DFD_SETSIZE=#{options[:fd_setsize]}"
```

Comment:
Is `-DDARWIN_UNLIMITED_SELECT` not needed as well?

Reply:
Since I only tested it with `--poll`, instead of `select` as the default, most likely yes. I'll do some tests now.

Reply:
Without `--poll` and with `--fd-setsize 2048`, I had success in getting the correct `ulimit -n` and the starting processes test. I did the same with `-DDARWIN_UNLIMITED_SELECT` and same result, it worked.

With that in mind, I could add it to the changeset, since it appears to be 'harmless'?